internal/isoeditor/rhcos.go

```diff
@@ -95,9 +106,48 @@ func formatRootfsServiceConfigFile(clusterProxyInfo *ClusterProxyInfo) (string,
 }
 
 func addFileToArchive(w *cpio.Writer, path string, content string, mode cpio.FileMode) error {
+	err := addDirToArchive(w, path)
+	if err != nil {
+		return err
+	}
+	hdr := &cpio.Header{
+		Name: path,
+		Mode: mode,
+		Size: int64(len(content)),
+	}
+	if err := w.WriteHeader(hdr); err != nil {
+		return err
+	}
+	if _, err := w.Write([]byte(content)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func addSymlinkToArchive(w *cpio.Writer, path string, target string) error {
```

Comment:
I don't think this is necessary unless I'm missing some nuance.

Couldn't we pass the target as the content and change the mode using the existing function? Why add a new one?

Reply:
It isn't necessary, I willl remove it.